client/db/src/upgrade.rs

```diff
@@ -83,8 +83,11 @@ fn migrate_2_to_3<Block: BlockT>(db_path: &Path, _db_type: DatabaseType) -> sp_b
 	let mut transaction = db.transaction();
 	for key in keys {
 		if let Some(justification) = db.get(columns::JUSTIFICATIONS, &key).map_err(db_err)? {
-			// Tag each Justification with the hardcoded ID for GRANDPA. Avoid the dependency on the GRANDPA crate
-			let justifications = sp_runtime::Justifications::from((*b"FRNK", justification));
+			// Tag each justification with the hardcoded ID for GRANDPA to avoid the dependency on
+			// the GRANDPA crate.
+			// NOTE: when storing justifications the previous API would get a `Vec<u8>` and still
+			// call encode on it.
+			let justifications = sp_runtime::Justifications::from((*b"FRNK", justification.decode()));
```

Comment:
This is off-topic, but I'm really surprised that `decode()` doesn't return an error.
Do we simply panic if for example the `Vec` says it has 5 elements but only 4 are found?

Reply:
No it returns an error :D 

@andresilva just had forgotten to handle it.